### PR TITLE
added readiness gate in specialization

### PR DESCRIPTION
**Pull request by henderiw**
added readiness gate in specialization
updated error handling and logging -> errors get added to the condition message
reorganised and removed obsolete code

### CODE REVIEW

File: krm-functions/lib/condkptsdk/condition.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package condkptsdk
+
+import (
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	kptv1 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1"
+	kptfilev1 "github.com/nephio-project/nephio/krm-functions/lib/kptfile/v1"
```
[the remark this answers is not in the document]
Comment (henderiw) [reply krm-functions/lib/condkptsdk/condition.go:21]: ok we use v1 as this is the version of the kptfile. If it ever changes we have the reference to the source. This is how it is done in other project. 
kptv1 is the porch repo
kptfilev1 or kptfiellibv1 is our additions to it as it is missing some convenience methods. Will align the import libs

[the remark this answers is not in the document]
Comment (henderiw) [reply krm-functions/lib/condkptsdk/condition.go]: aligned the import naming to kptfilelibv1

**Review by gvbalaji — APPROVED** [review]
/approve

**Pull request merged**